server/src/internalClusterTest/java/org/opensearch/cluster/shards/ClusterShardLimitIT.java

```diff
@@ -127,6 +133,69 @@ public void testIndexCreationOverLimit() {
         assertFalse(clusterState.getMetadata().hasIndex("should-fail"));
     }
 
+    public void testIndexCreationOverLimitForHiddenIndexesSucceeds() {
+        int dataNodes = client().admin().cluster().prepareState().get().getState().getNodes().getDataNodes().size();
+
+        ShardCounts counts = ShardCounts.forDataNodeCount(dataNodes);
+
+        setShardsPerNode(counts.getShardsPerNode());
+
+        createIndex(
+            "test",
+            Settings.builder()
+                .put(indexSettings())
+                .put(SETTING_NUMBER_OF_SHARDS, counts.getFirstIndexShards())
+                .put(SETTING_NUMBER_OF_REPLICAS, counts.getFirstIndexReplicas())
+                .build()
+        );
+
```

Comment:
It would be good to add assertions around the max shard per node setting value and the total number of shards.

Reply:
ACK will add